src/app/pages/login/login-form/login-form.component.spec.ts

```diff
@@ -0,0 +1,132 @@
+        },
+        message: "You have successfully logged in",
+        status: "success"
+    }
+}
+describe('LoginFormComponent', () => {
+  let component: LoginFormComponent;
+  let fixture: ComponentFixture<LoginFormComponent>;
+  
+  beforeAll(() => resetSpies([loginServiceSpy]));
+  afterEach(() => resetSpies([loginServiceSpy]));
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ReactiveFormsModule,
+        FormsModule,
+        HttpClientTestingModule,
+        RouterTestingModule,
+        ToastrModule.forRoot()
+      ],
+      declarations: [
+        LoginFormComponent,
+        LoginHeaderComponent,
+        LoginSliderComponent,
+        SocialLoginComponent,
+      ],
+      providers: [{provide: LoginService, useValue: loginServiceSpy},
+        {provide: ToastrService, useValue: toastServiceSpy}]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(LoginFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should login a user successfully', () => {
+    const { email, password } = component.loginForm.controls;
+    email.setValue('testemail@email.com');
+    password.setValue('password');
+    loginServiceSpy.login.and.returnValue(of(loginResponse));
+    component.onLogin(loginData);
+    expect(toastServiceSpy.success).toHaveBeenCalledWith(loginResponse.data.message);
+  })
+
+  it('should throw an error in case of wrong details', () => {
+    const { email, password } = component.loginForm.controls;
+    email.setValue('testemail@email.com');
+    password.setValue('password');
+    loginServiceSpy.login.and.returnValue(throwError(
+      loginError
+    ));
+    component.onLogin(loginData);
+    expect(toastServiceSpy.error).toHaveBeenCalledWith('Invalid email and password combination');
+  })
+  it('should call login method', () => {
+    component.onLogin(loginData)
+    expect(component.success).toBeFalsy()
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should be invalid if empty', () => {
+    expect(component.loginForm.valid).toBeFalsy();
+  });
+
+  it('should have invalid email field if empty', () => {
+    const email = component.loginForm.controls.email;
+    expect(email.valid).toBeTruthy();
+  });
+
+  it('should be valid for email if email is passed', () => {
+    const email = component.loginForm.controls.email;
+    email.setValue('someemail@email.com');
+    expect(email.valid).toBeTruthy();
+  });
+  it('tests something async', function(done) {
+    const { email, password } = component.loginForm.controls;
+    email.setValue('testemail@email.com');
+    password.setValue('password');
+    setTimeout(function() {
+        expect(component.inputError).toBeFalsy()
+        done();
+    }, 3000);
+});
+  it('an invalid form passed', () => {
```

Comment:
@BrianSerem, thanks for the good work, However it's customary that 'it' is followed by 'should' in your test descriptions, How you used it on line 99 can be a great good example.

Reply:
Yes sure, I have renamed the test description to contain the word 'should'

Reply:
Good job